### ISSUE TITLE
Errors when specifying complex note name directories

### ISSUE THREAD

**Issue by technotherion** [bartkessels/daily-note-calendar#11]
My Periodic Notes format for my daily notes is as following in the Periodic Notes plugin: 
"YYYY/[Daily]/MM/YY-MM-DD" where [Daily] is exact text. Today's daily note would be 2024/Daily/11/2024-11-16. 
In Periodic Notes, this works as intended and I have tried to replicate these settings as close as possible within Daily Note Calendar. 

When following date-fns syntax, I end up with "yyyy/[Daily]/MM/yyyy-MM-dd". When clicking on one of the day boxes in the calendar, I get an error in the console.
```
plugin:daily-note-calendar:13 Uncaught (in promise) RangeError: Format string contains an unescaped latin alphabet character `l`
```

When removing "Daily" from the filename, I'm left with "yyyy/MM/yyyy-MM-dd". I try again to click another date box on the calendar and get the following errors:

```
Uncaught (in promise) Error: ENOENT: no such file or directory, copyfile 'C:\[My Vault Path]\_templates\Daily.md' -> 'C:\[My Vault Path]\0. PeriodicNotes\2024\11\2024-11-22.md'
```
The specified path for the template does exist and the latter file is never created. 

Simplifying things further... narrowing it to "yyyy-MM-dd" does allow for the file to be successfully created, although not in the directory I'm wanting it to be created in. 

Is there a way that I'm supposed to be specifying a file name with a mixture of dates and directories? I can provide further examples of how my daily/periodic notes are set up and how it's supposed to function if it is at all helpful to you.

**Comment by bartkessels** [comment]
Hello, thanks for your feedback.

For the date-fns syntax to add a hard-coded string in the name, you'd need the `'` single-quote as an escape character. So, in your case that would be `yyyy/'Daily'/MM/yyyy-MM-dd`. But this indeed leads to another failure in the plugin, which is that it cannot create directories in the vault.

So, both of these problems can be fixed in the plugin and to fix the first problem I think I'm going to add a preview in the settings menu with the most common characters to create a name template. As for the second issue, I'm going to search for a solution.

To sum it up, I'll be creating a release which allows you to see in the settings menu what characters are available to build up your name or folder template, and give you a live preview. And fix the bug that won't allow you to create a note if the folder doesn't exist, by creating the required folders.

I hope this will help you out, if not feel free to reply to this issue. For now, keep the feedback coming and in an upcoming release (probably 1.5.0) this issue will be resolved!

**Comment by bartkessels** [comment]
This issue is resolved in PR https://github.com/bartkessels/daily-note-calendar/pull/12 and will be released with version 1.4.0.

Thanks again for your feedback, and if you have anymore issues or feature-requests, keep them coming!

**Comment by technotherion** [comment]
Confirming that I've tested my Daily, Weekly, Monthly and Yearly notes and they all work perfectly! The Format Example section is extremely valuable, thank you for including it.
